### ISSUE TITLE
Restore overlay widget logic in States.js

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#2083]
@peller 
In branch StatesCleanup1, I have temporarily commented out some logic from the bottom of States.js that deals with overlay widgets. This issue is a reminder to restore that logic.

**Comment by peller** [comment]
changeset?

**Comment by JonFerraiolo** [comment]
Overlay widget logic in States.js was commented out in this commit https://github.com/maqetta/maqetta/commit/c2b0c42b9dbdd8aa8780a2e174169771f95acd50#diff-1 which is currently only in this branch https://github.com/maqetta/maqetta/commits/StatesCleanup1. I hope to merge into master today or tomorrow.

**Comment by aerwin** [comment]
@JonFerraiolo @peller 

Would this one explain why clicking on an overlay in the outline palette doesn't cause the overlay to become visible in my testing today?

**Comment by JonFerraiolo** [comment]
Probably!

**Comment by peller** [comment]
@aerwin, I want to say this regressed prior to M5 and is awaiting a redesign based on the new states code

**Comment by aerwin** [comment]
@peller Ok, if this is a known regression from awhile back then I won't open a new issue based on my testing today.

**Comment by JonFerraiolo** [comment]
-> @peller 

I looked at AppStates.js and concluded that all we need to do to restore the old logic at the bottom of the file is simply remove the comments and re-activate the code. I tried it out on an experimental basis and it looks to me like I didn't really need to comment out that code in the first place. (I had commented it out because I had my hands full with the major changes I was making and didn't want to deal with overlay widgets at that time.)

Reassigning to @peller because I suspect there are lots of other related regressions having to do with overlay widgets, figuring that I'll leave it to @peller to study the code I'm referring to at the bottom of AppStaets.js, (probably) remove the comments, and then check in, but letting him do it in case he sees the need for other associated changes.

Remember, both versions of AppStates.js will need changes.
